backfill/2020-03-18-gcs-errors/README.md

```diff
@@ -0,0 +1,116 @@
+We had our BQ live sinks write a number of records to error output in GCS between
+2020-03-16 and 2020-03-18 while we were experiencing pipeline instability.
+There were a couple of bad things that worked together to cause this:
+
+- Our error handling code around fetching table info from the BigQuery API was
+  incorrect; it should have bubbled up and shut down the worker, but instead
+  caught the exceptions and wrote them silently to error output
+- We started exceeding the API rate limit regularly on 2020-03-16 because
+  we scaled up the structured sink to more workers that we had used before
+  and we didn't realize that we were already close to hitting the API limit
+  
+This affected both structured and telemetry pipelines.
+
+For structured, we want to backfill from errors for the affected days
+2020-03-16 through 2020-03-18.
+
+For telemetry, we also had Dataflow jobs failing in this time period, so we
+had to do some backfills from payload_bytes_decoded, which also gets the records
+that would have been recovered from this process. So, for telemetry we only need
+to hit 2020-03-16.
+
+## Stage GCS objects
+
+Dataflow in the backfill project got permission denied attempting to read
+directly from the prod error data, so let's copy the objects to a bucket
+in the backfill project.
+
+First, we need to create the bucket (done manually in the BQ console).
+
+Now, structured:
+
+```
+for x in 2020-03-16 2020-03-17 2020-03-18
+do 
+  gsutil -m cp -r gs://moz-fx-data-prod-data/structured-decoded_bq-sink/error/${x}/ gs://moz-fx-data-backfill-data/structured-decoded_bq-sink/error/
```

Comment:
Just a note for future backfills, I think we should probably use a bucket with a unique name to avoid accidentally munging data together.

Reply:
Agreed. It was a mistake that I didn't include the number when creating this bucket.